src/sentry/integrations/slack/requests/event.py

```diff
@@ -76,15 +96,20 @@ def _validate_event(self) -> None:
     def _validate_integration(self) -> None:
         super()._validate_integration()
 
-        if self.text and self.text in COMMANDS:
```

Comment:
I think this would be cleaner as a regular conditional
```python
if (
	(self.text and self.text in COMMANDS) or 
	(self.type == "link_shared" and has_discover_links(self.links))
):
```